exporters/committime/__init__.py

```diff
@@ -91,3 +95,31 @@ def __parse_repourl(self):
         self.__repo_group = parsed.owner
         self.__repo_name = parsed.name
         self.__repo_project = parsed.name
+
+
+def commit_metric_from_build(app: str, build, errors: list) -> CommitMetric:
+    """
+    Create a CommitMetric from build information.
+    Will collect errors for missing data instead of failing early.
+    """
+    # set attributes based on a mapping from attribute name to
+    # lookup path.
+    # Collect all errors to be reported at once instead of failing fast.
+    metric = CommitMetric(app)
+    for attr_name, path in _COMMIT_METRIC_MAPPING.items():
+        with collect_bad_attribute_path_error(errors):
+            value = get_nested(build, path, name="build")
+            setattr(metric, attr_name, value)
+
+    return metric
+
+
+_COMMIT_METRIC_MAPPING = dict(
+    build_name="metadata.name",
+    build_config_name="metadata.labels.buildconfig",
+    namespace="metadata.namespace",
+    commit_hash="spec.revision.git.commit",
+    committer="spec.revision.git.author.name",
+    image_location="status.outputDockerImageReference",
+    image_hash="status.output.to.imageDigest",
+)
```

Comment:
I am missing some metrics removed from https://github.com/konveyor/pelorus/pull/411/files#diff-2d6d20dd9681db562d438793a6b1730a740965526f49c4b7b34fbdb8d0fa354bL184-L196

Those are ones which I found missing: repo_url, labels, name, commit_timestamp

Could you explain why they are out ?

Could this be moved to the top of the file? Easier to read such mappings when they are declared in one place (below SUPPORTED_PROTOCOLS).

Reply:
Alternatively, for labels, I could make a custom setter. But that seems like more work than necessary.

Reply:
Yep, let's keep it simple.

Class attribute sounds fine.

For the rest a little doc would be sufficient, so we know later why it was done this way.